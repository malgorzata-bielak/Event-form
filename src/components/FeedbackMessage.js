import React from "react";
import styled from "@emotion/styled";

const ValidationError = styled.p`
  margin: 0 0 25px;
  color: #d41b56;
  font-style: italic;
`;

const Response = styled.p`
  margin: 0 0 25px;
  font-style: italic;
`;

const FeedbackMessage = ({ validationError, isSaving, databaseResponse }) => (
  <>
    <ValidationError>{validationError}</ValidationError>
    <Response>{databaseResponse}</Response>
    {isSaving && <Response>Please wait...</Response>}
  </>
);

export default FeedbackMessage;
