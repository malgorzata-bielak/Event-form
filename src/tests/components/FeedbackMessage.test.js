import React from "react";
import { shallow } from "enzyme";

import FeedbackMessage from "../../components/FeedbackMessage";

const propsData = [
  { validationError: "", isSaving: false, databaseResponse: "" },
  {
    validationError: "Please provide all the necessary information",
    isSaving: false,
    databaseResponse: ""
  },
  { validationError: "", isSaving: true, databaseResponse: "" },
  {
    validationError: "",
    isSaving: false,
    databaseResponse: "Your data has been saved successfully"
  }
];

test("should render FeedbackMessage with default values", () => {
  const wrapper = shallow(<FeedbackMessage {...propsData[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render validation error message", () => {
  const wrapper = shallow(<FeedbackMessage {...propsData[1]} />);

  const paragraph = wrapper
    .find("Styled(p)")
    .at(0)
    .text();
  const value = "Please provide all the necessary information";
  expect(paragraph).toEqual(value);
});

test("should render saving state message", () => {
  const wrapper = shallow(<FeedbackMessage {...propsData[2]} />);

  const paragraph = wrapper
    .find("Styled(p)")
    .at(2)
    .text();
  const value = "Please wait...";
  expect(paragraph).toEqual(value);
});

test("should render database response", () => {
  const wrapper = shallow(<FeedbackMessage {...propsData[3]} />);

  const paragraph = wrapper
    .find("Styled(p)")
    .at(1)
    .text();
  const value = "Your data has been saved successfully";
  expect(paragraph).toEqual(value);
});
