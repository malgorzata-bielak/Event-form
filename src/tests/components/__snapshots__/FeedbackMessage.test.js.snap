// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FeedbackMessage with default values 1`] = `
<Fragment>
  <Styled(p) />
  <Styled(p) />
</Fragment>
`;
